### DISCUSSION TITLE
Modbus raw hex values to mqtt

### DISCUSSION THREAD

**Discussion by nikito7** [arendst/Tasmota#19545] [Q&A] [answered]
Can I send raw hex values via mqtt?

I have some string registers or mixed registers.

Thank you

**Comment by nikito7** [comment]
@gemu2015

**Reply by gemu2015** [reply]
do you mean in SML driver ? MODBUS string registers are not supported in SML.
they will give a decimal value which can be converted to hex and then sent to MQTT

**Reply by nikito7** [reply]
Yes sml.

I need in hex: 2 bytes, 4 bytes, 6 bytes, or more.

Any working mask?

Something like

010440xxHHHHxxxx@...
010440xxHHHHHHxxxx@...

Thank you

**Reply by gemu2015** [reply, marked as answer]
there is no HH mask
you must use unsigned word, long etc.  e.g. UUuu  then convert to an hex string
however when you use normal scripting you have single precision float only which means you must stick to 16 bit unsigned registers 
and concatenate the results if you need longer hex values.

**Comment by nikito7** [comment]
Thank you
